lib/bom.js

```diff
@@ -261,6 +261,61 @@ declare class Worker extends EventTarget {
     terminate(): void;
 }
 
+declare class SharedWorker extends EventTarget {
+    constructor(stringUrl: string): void;
+    port: MessagePort;
+    onerror: (ev: Event) => any;
+}
+
+declare function importScripts(...urls: Array<string>): void;
+
+declare class WorkerGlobalScope extends EventTarget {
+    self: WorkerGlobalScope;
+    location: WorkerLocation;
+    navigator: WorkerNavigator;
+    importScripts(...urls: Array<string>): void;
+    onerror: (ev: Event) => any;
+    onlanguagechange: (ev: Event) => any;
+    onoffline: (ev: Event) => any;
+    ononline: (ev: Event) => any;
+    onrejectionhandled: (ev: PromiseRejectionEvent) => any;
+    onunhandledrejection: (ev: PromiseRejectionEvent) => any;
+}
+
+declare class DedicatedWorkerGlobalScope extends WorkerGlobalScope {
+    close(): void;
+    onmessage(): (ev: MessageEvent) => any;
+    postMessage(message: any, transfer?: Iterable<Object>): void;
+}
+
+declare class SharedWorkerGlobalScope extends WorkerGlobalScope {
+    close(): void;
+    name: string;
+    onconnect: (ev: MessageEvent) => any;
+}
+
+declare class WorkerLocation {
+    origin: string;
+    protocol: string;
+    host: string;
+    hostname: string;
+    port: string;
+    pathname: string;
+    search: string;
+    hash: string;
+}
+
+declare class WorkerNavigator {
```

Comment:
Hey, would it be possible to factor these props into (say) `NavigatorCommon` and then have both this and `Navigator` extend it? It'd be cumbersome to actually match the spec and break them up into several different superinterfaces, but a single block of the common properties should be simple enough, I think... 

LMK if I'm missing a subtlety though - don't want to make things harder than they need to be. :)